modules/yieldlabBidAdapter.js

```diff
@@ -441,4 +445,31 @@ function extractSizes(bid) {
   return Array.from(deduplicatedSizeStrings)
 }
 
+/**
+ * Gets the floor price if the Price Floors Module is enabled for a given auction,
+ * which will add the getFloor() function to the bidRequest object.
+ *
+ * @param {Object} bid
+ * @param {string[]} sizes
+ * @returns The floor CPM of a matched rule based on the rule selection process (mediaType, size and currency),
+ *          using the getFloor() inputs. Multi sizes and unsupported media types will default to '*'
+ */
+function getBidFloor(bid, sizes) {
+  if (!isFn(bid.getFloor)) {
+    return undefined;
+  }
+  const mediaTypes = deepAccess(bid, 'mediaTypes');
+  const mediaType = mediaTypes !== undefined && spec.supportedMediaTypes.includes(Object.keys(mediaTypes)[0].toLowerCase())
+    ? Object.keys(mediaTypes)[0].toLowerCase() : '*';
+  const floor = bid.getFloor({
+    currency: CURRENCY_CODE,
+    mediaType: mediaType,
+    size: sizes.length !== 1 ? '*' : extractSizes(sizes)
+  });
+  if (isPlainObject(floor) && !isEmpty(floor) && !isNaN(floor.floor) && floor.currency === CURRENCY_CODE) {
```

Comment:
I think

```javascript
if (floor.currency === CURRENCY_CODE) {
  ...
}
```

should be enough. The docs say that `getFloor()` returns either

```
{
  currency: string,
  floor: float
}
```

or the empty object `{}` if a floor could not be found for the given input.

Reply:
I'm worried that it is possible that the currency is not the same. From the [documentation](https://docs.prebid.org/dev-docs/modules/floors.html#getfloor-response:~:text=If%20the%20Floor%20Module%20cannot%20convert%20a%20floor%20to%20the%20supplied%20currency%2C%20bid%20adapters%20will%20be%20required%20to%20handle%20the%20supplied%20floor.):
_If the Floor Module cannot convert a floor to the supplied currency, bid adapters will be required to handle the supplied floor._

Reply:
I just saw that other adapters do it the same way. So I guess it's fine.